### DISCUSSION TITLE
Guidance on Integrating Custom RAG Chat Engine (llama-index) with Lobe Chat

### DISCUSSION THREAD

**Discussion by smendig** [lobehub/lobehub#1507] [General Question]
Hi Lobe Chat Community,

I'm looking to integrate a custom Retrieval-Augmented Generation (RAG) chat engine developed with llamaindex into Lobe Chat, aiming to utilize its UI for a better user experience. After reviewing the [Chat API documentation](https://github.com/lobehub/lobe-chat/wiki/Chat-API) on the Lobe Chat wiki, the documentation is clear on integrating with OpenAI, but I'm at a standstill on how to proceed with a custom engine like mine.

In essence, here's what I've got:

```
chat_engine = index.as_chat_engine()
response = chat_engine.chat("Search about whatever.")
```

I'm reaching out for insight on adapting Lobe Chat to work with a custom chat engine, like llama-index. Any advice or pointers would be greatly appreciated.

Thanks in advance for any help you can offer!

**Comment by arvinxx** [comment]
By now we don't support custom RAG, but. we will work on this this year! stay tuning for it!

**Reply by BrandonStudio** [reply]
Do you think this can be done via plugin system?

**Reply by arvinxx** [reply]
Yes And I know there are some users do it in plugin way. But I prefer to support it in LobeChat natively.

**Comment by oldgithubman** [comment]
Built-in RAG would be huge

**Comment by AxDSan** [comment]
was this implemented? does LobeChat offers RAG?
